controllers/admin/AdminOrdersController.php

```diff
@@ -478,19 +478,64 @@ public function postProcess()
                         // Send mail to customer
                         $customer = new Customer((int)$order->id_customer);
                         $carrier = new Carrier((int)$order->id_carrier, $order->id_lang);
+                        $address = new Address((int)$order->id_address_delivery);
+
                         if (!Validate::isLoadedObject($customer)) {
                             throw new PrestaShopException('Can\'t load Customer object');
                         }
                         if (!Validate::isLoadedObject($carrier)) {
                             throw new PrestaShopException('Can\'t load Carrier object');
                         }
+                        if (!Validate::isLoadedObject($address)) {
+                            throw new PrestaShopException('Can\'t load Address object');
+                        }
+
+                        $products = $order->getCartProducts();
+
+                        $metadata = '';
+
+                        $link = Context::getContext()->link;
+
+                        foreach ($products as $product) {
+                            $metadata .= '<div itemprop="itemShipped" itemscope itemtype="http://schema.org/Product">';
```

Comment:
@pierreavizou hard coding this would make it impossible to create other types of emails I think? See https://developers.google.com/gmail/markup/reference/order under the section order with billing details. The reference would become **itemOffered**